packages/web/src/store/slices/websiteResults.ts

```diff
@@ -0,0 +1,75 @@
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import type { WebPage, WebPagePackage, PackageVulnerabilityData } from '@gradejs-public/shared';
+import { DefaultFiltersAndSorters, FiltersState } from '../../components/layouts/Filters/Filters';
+import { client } from '../../services/apiClient';
+import { trackCustomEvent } from '../../services/analytics';
+
+const defaultDetectionResult = {
+  packages: [] as WebPagePackage[],
+  vulnerabilities: {} as Record<string, PackageVulnerabilityData[]>,
+  webpages: [] as WebPage[],
+};
+
+const initialState = {
+  filters: { ...DefaultFiltersAndSorters },
+  isFailed: false,
+  isLoading: false,
+  detectionResult: defaultDetectionResult,
+};
+
+const sleep = (ms: number | undefined) =>
+  new Promise((r) => {
+    setTimeout(r, ms);
+  });
+
+const hasPendingPages = (result: DetectionResult) =>
+  !!result.webpages.find((item) => item.status === 'pending');
+
+const getWebsite = createAsyncThunk('websiteResults/getWebsite', async (hostname: string) => {
+  const loadStartTime = Date.now();
+  await client.mutation('syncWebsite', hostname);
+  let results = await client.query('getWebsite', hostname);
+  while (hasPendingPages(results)) {
+    await sleep(5000);
+    await client.mutation('syncWebsite', hostname);
+    results = await client.query('getWebsite', hostname);
```

Comment:
Could you clarify why would we need two separate requests in here?

Reply:
Changed this, but I think we should take a look at it again, it still looks weird.

Reply:
Looks passable to me for now, we'll revisit it with redesign impl